operator/duties/attester.go

```diff
@@ -0,0 +1,261 @@
+			// If we have reached the mid-point of the epoch, fetch the duties for the next epoch in the next slot.
+			// This allows us to set them up at a time when the beacon node should be less busy.
+			if uint64(slot)%slotsPerEpoch == slotsPerEpoch/2-2 {
+				h.fetchNextEpoch = true
+			}
+
+			// last slot of epoch
+			if uint64(slot)%slotsPerEpoch == slotsPerEpoch-1 {
+				h.duties.Reset(currentEpoch)
+			}
+
+		case reorgEvent := <-h.reorg:
+			currentEpoch := h.network.Beacon.EstimatedEpochAtSlot(reorgEvent.Slot)
+			buildStr := fmt.Sprintf("e%v-s%v-#%v", currentEpoch, reorgEvent.Slot, reorgEvent.Slot%32+1)
+			h.logger.Info("🔀 reorg event received", zap.String("epoch_slot_seq", buildStr), zap.Any("event", reorgEvent))
+
+			// reset current epoch duties
+			if reorgEvent.Previous {
+				h.duties.Reset(currentEpoch)
+				h.fetchFirst = true
+				h.fetchCurrentEpoch = true
+			} else if reorgEvent.Current {
+				// reset & re-fetch next epoch duties if in appropriate slot range,
+				// otherwise they will be fetched by the appropriate slot tick.
+				if h.shouldFetchNexEpoch(reorgEvent.Slot) {
+					h.duties.Reset(currentEpoch + 1)
+					h.fetchNextEpoch = true
+				}
+			}
+
+		case <-h.indicesChange:
+			slot := h.network.Beacon.EstimatedCurrentSlot()
+			currentEpoch := h.network.Beacon.EstimatedEpochAtSlot(slot)
+			buildStr := fmt.Sprintf("e%v-s%v-#%v", currentEpoch, slot, slot%32+1)
+			h.logger.Info("🔁 indices change received", zap.String("epoch_slot_seq", buildStr))
+
+			h.indicesChanged = true
+			h.fetchCurrentEpoch = true
+
+			// reset next epoch duties if in appropriate slot range
+			if h.shouldFetchNexEpoch(slot) {
+				h.duties.Reset(currentEpoch + 1)
+				h.fetchNextEpoch = true
+			}
+		}
+	}
+}
+
+func (h *AttesterHandler) processFetching(ctx context.Context, epoch phase0.Epoch, slot phase0.Slot) {
+	ctx, cancel := context.WithDeadline(ctx, h.network.Beacon.GetSlotStartTime(slot+1).Add(100*time.Millisecond))
+	defer cancel()
+
+	if h.fetchCurrentEpoch {
+		if err := h.fetchAndProcessDuties(ctx, epoch); err != nil {
+			h.logger.Error("failed to fetch duties for current epoch", zap.Error(err))
+			return
+		}
+		h.fetchCurrentEpoch = false
+	}
+
+	if h.fetchNextEpoch && h.shouldFetchNexEpoch(slot) {
+		if err := h.fetchAndProcessDuties(ctx, epoch+1); err != nil {
+			h.logger.Error("failed to fetch duties for next epoch", zap.Error(err))
+			return
+		}
+		h.fetchNextEpoch = false
+	}
+}
+
+func (h *AttesterHandler) processExecution(epoch phase0.Epoch, slot phase0.Slot) {
+	// range over duties and execute
+	if slotMap, ok := h.duties.m[epoch]; ok {
+		if duties, ok := slotMap[slot]; ok {
+			toExecute := make([]*spectypes.Duty, 0, len(duties)*2)
+			for _, d := range duties {
+				if h.shouldExecute(d) {
+					toExecute = append(toExecute, h.toSpecDuty(d, spectypes.BNRoleAttester))
+					toExecute = append(toExecute, h.toSpecDuty(d, spectypes.BNRoleAggregator))
+				}
+			}
+			h.executeDuties(h.logger, toExecute)
+		}
+	}
```

Comment:
this might be just code-style but I prefer less nested code. feel free to dismiss..
```suggestion
	slotMap, ok := h.duties.m[epoch];
	if !ok
	   return
	}
	duties, ok := slotMap[slot]
	if !ok {
	    return
	}
	toExecute := make([]*spectypes.Duty, 0, len(duties)*2)
	for _, d := range duties {
		if h.shouldExecute(d) {
			toExecute = append(toExecute, h.toSpecDuty(d, spectypes.BNRoleAttester))
			toExecute = append(toExecute, h.toSpecDuty(d, spectypes.BNRoleAggregator))
		}
	}
	h.executeDuties(h.logger, toExecute)
```